stage1/init/common/mount.go

```diff
@@ -294,8 +296,29 @@ func AppAddMounts(p *stage1commontypes.Pod, ra *schema.RuntimeApp, enterCmd []st
 		os.Exit(254)
 	}
 
+	absRoot, err := filepath.Abs(p.Root)
+	if err != nil {
+		log.FatalE("could not determine pod's absolute path", err)
+	}
+
+	appRootfs := common.AppRootfsPath(absRoot, ra.Name)
+
 	for _, m := range mounts {
-		AppAddOneMount(p, ra, m.Volume.Source, m.Mount.Path, m.ReadOnly, enterCmd)
+		shPath := filepath.Join(common.SharedVolumesPath(p.Root), m.Volume.Name.String())
+		mntPath, err := EvaluateSymlinksInsideApp(appRootfs, m.Mount.Path)
+		if err != nil {
+			log.Fatalf("Could not evaluate path %v: %v", m.Mount.Path, err)
+		}
+		mntAbsPath := filepath.Join(appRootfs, mntPath)
+		// Create the stage1 destination
+		if err := PrepareMountpoints(shPath, mntAbsPath, &m.Volume, m.DockerImplicit); err != nil {
+			log.FatalE("could not prepare mountpoint", err)
+		}
+		source := m.Volume.Source
+		if m.Volume.Kind == "empty" {
```

Comment:
might as well use a switch statement like in https://github.com/coreos/rkt/blob/a252c4c3078f746eca2739f3560428aa59b89da2/stage1/init/common/pod.go#L522, right?